submissions/nataliaArtsiukh/Frogger-game/app.js

```diff
@@ -0,0 +1,141 @@
+var colWidth = 101;
+var rowHeight = 83;
+
+var Star = function(numCol) {
+    this.x = colWidth * numCol;
+    this.y = -10;
+    this.sprite = 'images/Star.png';
+}
+
+Star.prototype.update = function() {
+    if (player.y < 0) {
+        var index = player.x / 101;
+        allStars[index] = null;
+        if (allStars.filter(elem => elem).length == 0) {
+            setTimeout(()=> {
+                alert('You are amazing!!!'); 
+                launch();
+            }, 1000);
+        } 
+    }
+}
+
+Star.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+}
+
+// Enemies our player must avoid
+
+var Enemy = function(numRow) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.x = colWidth * -1;
+    this.y = rowHeight * numRow - 20;
+    this.speed = (Math.random() + 0.7) * colWidth;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    if (this.x < colWidth * 5) {
+        this.x = this.x + this.speed * dt;
+    } else {
+        this.x = colWidth * -1;
+        this.speed = (Math.random() + 0.7) * colWidth;
+    }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+var Player = function() {
+    this.x = colWidth * 2;
+    this.y = rowHeight * 5 - 10;
+    this.finished = false;
+    this.sprite = 'images/char-boy.png';
+    this.update = function(){
+        for (var i = 0; i < allEnemies.length; i++) {
+            if (this.y == allEnemies[i].y + 10) {
+                if (allEnemies[i].x > this.x - colWidth + 20 && allEnemies[i].x < this.x + colWidth - 20) {
+                    this.x = colWidth * 2;
+                    this.y = rowHeight * 5 - 10;
+                } 
+            }
+        }
```

Comment:
(1) Use `Array` methods whenever possible, in general.
(2) Probably it makes more chance that it is an enemy checks if they collided with the Player, rather than Player scans all enemies.
(3) Classes should not refer to any global variables, like global variable `player` or `enemies`, which is an instance of `Player` class or contains instances of `Enemy` class.

(referring to global constants and globals provided by the gaming platform like `Resources` is OK);

Hint: you can pass `Player` instance as an argument to every enemy.

(4) Separation of Concerns principle should be followed.  (e.g. `update` method does only rendering and doesn't contain any unrelated **inline** code; for example collision check is defined as a dedicated method and only called from inside `update`)